### PR TITLE
Documentation: Improve 'On ideologically motivated changes'

### PR DESCRIPTION
**Pull request by n0toose**
From my previous interactions with the Serenity community, it was clear to me that this rule (for better or worse) was introduced by the author that wanted to focus on the code and not let an open-source operating system project be used as a 'political vehicle' and so as to not have to possess a political science degree to operate the project thought to be a technical exercise, like building a sandcastle.

For better or worse.

Unfortunately, the wording as-is can be interpreted as a dogwhistle in the direction of "keep politics out of tech", which has been present in communities like, as of recently, NixCPP - and this seems to have caused the problems that Serenity has been intentionally trying to avoid.

It might also disregard cases of technical arguments, arguments involving how people treat each other _within_ the community as well as how to change the wording in documentation with the sole intent of making the project more attractive to more contributors.

Given recent commits and governance changes, I decided to rewrite the rule to make it more clear and encourage people to be "more excellent to each other", while not compromising on what I see as the original meaning.

### CODE REVIEW

**Comment by n0toose** [comment]
I understand that this subject might be very touchy, but I hope that it is understood that this change is made in good faith and that I just want to help (after seeing https://github.com/SerenityOS/serenity/pull/24648).

I tried to *explain* instead of leaving *lots* room for ambiguity - while still leaving some room - because as-is, the project unintentionally got itself involved in a *future* culture war. I modified the paragraph on how I interpreted the "why the drama from yesterday happened" - given that I understand both perspectives to some degree because of my previous (short) involvement here as well as in other projects. I was mildly dissatisfied with the way the rule was practically enforced in the past despite understanding its nature (I'm only trying to explain its nature), and I also believe that some people took it too far with the accusations against an entire community, **especially** given that the "ex-BDFL"-author is not involved anymore.

My goal here is to help, and I am definitely open to a discussion if you folks wish to have one - after all, I have not been involved in a while and my interpretation may be a bit one-sided.

Best,
:^)

**Comment by thiemowmde** [comment]
Exclusively addressing males is an ideology I do not support.

**Comment by n0toose** [comment]
> Exclusively addressing males is an ideology I do not support.

I'm on your side and have to say that this is not the intention - this is intended to be a continuation in the spirit Ryan's change, they are non-binary. I believe I have to make it clear that I am an **outsider** here despite the "Contributor" tag and do not represent the project.

**Comment by n0toose** [comment]
There are lots of people in the SerenityOS community - some of them are friends of mine - that are non-binary or started identifying as queer during their involvement. I find the vibe by the ex-maintainer from 3 years ago bad - and it's among the reasons why I stopped being involved back then, have very strong opinions on the subject and insist on using gender-neutral language in other projects - I find that explicitly associating using the singular "they" with politics to be political.

The aforementioned Pull Request changed what you just brought up, @thiemowmde. The question is "How do we all learn from a bad situation and move on from here?". What do you think?

**Comment by fogti** [comment]
@thiemowmde In regards to e.g. #24647, afaik most, if not all of the problematic cases were fixed by #24648 (https://github.com/SerenityOS/serenity/commit/a2a6bc534868773b9320ec3ca7399283cf7a375b)

**Comment by thiemowmde** [comment]
While I agree the proposed rewording is much better, the core message is largely the same as before. (Sorry, I understand this is the idea of this pull request.) Particularly, what does "purely technical" mean? Do the project owners have an agreed on understanding of what they consider a "culture war", when it's "external", and what it means to be "external"?

Documentation addresses real people. Writing inclusive documentation is an art form. It's certainly not "purely technical" and something a project can "reject" just because "we" (who exactly?) "feels" (feels what?) a change might "be sensitive" (to whom?).

@fogti I have seen that, thanks a lot!

**Comment by gmta** [comment]
Hi @n0toose ! Thanks for thinking with us on the wording, very much appreciated. We're currently in process of reviewing how our policies are effectively worded, which can definitely be improved. We'll get back to this PR soon.

**Comment by n0toose** [comment]
> While I agree the proposed rewording is much better, the core message is largely the same as before. (Sorry, I understand this is the idea of this pull request.) Particularly, what does "purely technical" mean? Do the project owners have an agreed on understanding of what they consider a "culture war", when it's "external", and what it means to be "external"?

> Documentation addresses real people. Writing inclusive documentation is an art form. It's certainly not "purely technical" and something a project can "reject" just because "we" (who exactly?) "feels" (feels what?) a change might "be sensitive" (to whom?).

OK, now that is something something that I find constructive and that I (and others) can absolutely work with. I am not sure if I agree on "the core message being largely the same as before" (because some interpreted it as a dogwhistle meaning that "only straight heterosexual males should contribute btw your existence is political lol" - and I 100% would interpret it this way myself if I was not involved).

Despite aiming for some "ambiguity", you're right that my sentence is very unclear. All of these points are 100% valid and should be addressed - the cause of the problem was that I was also trying not to make the changes excessively long so as to not scare maintainers (that are possibly on high alert) away and increase the chance of getting "one step further" and not cause even more disruption as an outsider.

**Comment by n0toose** [comment]
> We're currently in process of reviewing how our policies are effectively expressed, which can definitely be improved.

@gmta, keep in mind that this PR was created with what I perceived to be the current status quo in mind. You're free to modify it (or just let me know if I can help), but I have a feeling that @thiemowmde's feedback here may actually help out.

Have a great Wednesday!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -135,4 +135,10 @@ To make this easier, we do appreciate it if folks enable the "Allow edits from m
 
 ## On ideologically motivated changes
 
-This is a purely technical project. As such, it is not an appropriate arena to advertise your personal politics or religious beliefs. Any changes that appear ideologically motivated will be rejected.
+Serenity's goal is to enable collaboration between as many groups as *reasonably* possible, and we welcome contributions that make the project more accessisble to people.
```
Comment (kleinesfilmroellchen) [inline CONTRIBUTING.md:138]: accessisble -> accessible

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -135,4 +135,10 @@ To make this easier, we do appreciate it if folks enable the "Allow edits from m
...
+
+However, Serenity is intended to be a purely technical exercise, in the sense of it not being meant to invoke any sociopolitical change. We explicitly try to avoid drama or getting involved in "external" culture wars, and can reject changes that we feel to be sensitive.
```
Comment (kleinesfilmroellchen) [inline CONTRIBUTING.md:140]: feel to be -> consider to be

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -135,4 +135,10 @@ To make this easier, we do appreciate it if folks enable the "Allow edits from m
...
+
+For instance, we can reject changes promoting dogwhistles, religious beliefs—which are unambiguously off-topic—or real-world political candidates.
```
Comment (kleinesfilmroellchen) [inline CONTRIBUTING.md:142]: Em dash here looks a bit strange, not sure that we use that anywhere else. Consider putting the phrase in parentheses instead.

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
Purely stylistic improvements

File: CONTRIBUTING.md (hunk 1 above)
Comment (sdomi) [inline CONTRIBUTING.md:138]: - `groups` feels weird here. Are we thinking about different minorities, or different groups of people within the project (i.e. people who mostly work on Kernel, people who mostly work on userspace, etc)?
- `*reasonably*` can be misinterpreted too, because it's a personal viewpoint. Who's reason are we thinking about? IMHO either remove this word or make it more specific

I would rephrase this sentence as `Serenity's goal is to enable collaboration between as many folks as possible. We welcome any contributions making the project more accessible.`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -135,4 +135,10 @@ To make this easier, we do appreciate it if folks enable the "Allow edits from m
...
+
+We may miss the mark sometimes, but we encourage good-faith dialogue over anger.
```
Comment (sdomi) [inline CONTRIBUTING.md:144]: i feel like this sentence can be grossly misinterpreted if bad actors were to actually start contributing into the project. (i.e. "it's okay to be bad sometimes because everyone does that AND we have that in the rules!")

I'd reword this as `We're hoping to create a space where everyone is welcome to contribute. In a big project, that's not an easy task - we highly encourage good-faith dialogue over anger and assumption of guilt.` or something along those lines.

**Comment by sdomi** [comment]
the general view is good, and those changes highly reflect what I wanted to change in this document myself later down the line. but if those were to be accepted, we have to deeply think about how all of this can be read and misread, and also about taking a specific position as a community/project.

**Comment by n0toose** [comment]
I think that the recent feedback shared here seems to share a general view that I have to be more explicit about how I mean things - I wasn't 100% sure how to do it myself.

I will ultimately leave the change untouched for the time being as I believe that any progress that I'd make at this point in time has a high risk of being dismissed - given that there's an upcoming policy change that seems to be pending. I believe that the feedback of my attempt (which somewhat attempted to both-sides this issue) can still be helpful for the person that will pick this up at a later point in time and I totally like a constructive discussion from people with different perspectives taking place in public.

File: CONTRIBUTING.md (hunk 4 above)
Comment (n0toose) [reply CONTRIBUTING.md:144]: you're right, this leaves space for tone-policing

File: CONTRIBUTING.md (hunk 2 above)
Comment (iProgramMC) [reply CONTRIBUTING.md:140]: “Feel to be” is gramatically correct English.

Comment (sdomi) [reply CONTRIBUTING.md:140]: i'd argue in favour of what @kleinesfilmroellchen wrote, while `Feel to be` is correct english, `consider to be` has a clearer intent (for me, at least)

**Review by 9ary — COMMENTED** [review]
Definitely a step in the right direction. Thanks for pushing this forward!

This morning I was fairly disappointed in how these issues were handled in the past, but I'm glad changes are being made in this regard.

SerenityOS is a very cool project, keep up the good work. :heart:

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (n0toose) [reply CONTRIBUTING.md:140]: This change suggestion aligns with previous feedback - I'd say that technical writing or design work or translating technical terms is also "technical work" - otherwise I'd have used the word "engineering" or "engineering exercise", but it seems like this is just me.

[the remark this answers is not in the document]
Comment (9ary) [reply CONTRIBUTING.md:140]: I understand where you're coming from. My perspective is that working on such a big project involves collaboration, which is a social thing rather than technical. Contributors come here to (in no particular order):
- have fun
- collaborate
- build something cool

The phrase "purely technical" may come across as dismissive of the human aspects, possibly resulting in a sterile and corporate-like image. I hope this isn't the intent here. :P

[the remark this answers is not in the document]
Comment (n0toose) [reply CONTRIBUTING.md:140]: It's not, and I fully get where *you're* coming from - just tried to give background as to why I wrote this to begin with in the first place.

**Comment by Puppercino** [comment]
I'm not too experienced in Serenity OS but I have a great appreciation for how far this OS project has come. :) 

Basing on this PR conversation and the current version of the `On ideologically motivated changes` section, it feels like this is trying to solve something that might be better addressed by a Code of Conduct, in my opinion. 

It's pretty difficult to enforce a policy of no drama and no ideology, when disagreements are basically guaranteed when collaboration is happening at this scale. A Code of Conduct won't fully stop disagreements or drama, but my perspective is that it'll at least put guard rails in place, making clear how everyone should behave when contributing to Serenity.

**Comment by Puppercino** [comment]
@incrediblyimpressiveusername I understand that people would want contributions to be completely technical, though the second half of your comment sounds like you want Serenity to be a political vehicle for your homophobia but that's just my opinion. 😀

image

**Comment by gmta** [comment]
Please do not feed the trolls.

File: CONTRIBUTING.md (hunk 1 above)
Comment (milesrout) [reply CONTRIBUTING.md:138]: 'folks' is no different to 'groups'. A folk is a group, but it's just a very strange and archaic way to describe groups of people that suggests some kind of ethnic connection to most people. What about just leaving it as it is (it's fine as it is) or saying 'between as many people as possible'? Is Serenity's goal *actually* to 'enable collaboration between as many people as possible', though? Or is the goal to produce software? The original description seems accurate: it is a purely technical project aiming to produce a technical result.

File: CONTRIBUTING.md (hunk 4 above)
Comment (milesrout) [reply CONTRIBUTING.md:144]: 'Tone policing' is a good thing. The most important time to be calm and measured is when you are angry about something. These extra sentences should just not be added. They contribute nothing.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -135,4 +135,10 @@ To make this easier, we do appreciate it if folks enable the "Allow edits from m
 
 ## On ideologically motivated changes
```
[the remark this answers is not in the document]
Comment (milesrout) [reply CONTRIBUTING.md:136]: 'Inclusivity' is not neutral. It's a naked political buzzword.

File: CONTRIBUTING.md (hunk 4 above)
Comment (fogti) [reply CONTRIBUTING.md:144]: "Tone policing" is a bad thing, it encurages off-topic discussions while usually being vague enough to not being able to outright invoke a proper code of conduct discussion (or discarding inappropriate comments) or (default to) this minimal ethics primer discussed here.

**Comment by adroslice** [comment]
A more technical justification for the rejection of ideologically motivated **changes** in specific may be needed.
Regarding the PR from three years ago, the idea seems to be that it wouldn't have been okay for the wording to be neutral from the beginning, when this clearly *isn't the case*. The distinguishing factor, between creation and modification, is that it was an **isolated and targeted change that leads to speculation about the project leadership** no matter whether it gets accepted or not:
- If it gets *accepted*, valuable partners who take offense might give up on the project.
- If it gets *rejected*, valueable partners who take offense might give up on the project. Here, this outcome has occurred.

I believe it is this impossible problem that tipped the scales towards the PR being rejected, not on the merit of its contents, but on its *effect on the project as a whole* in our more polarized political climate.

It is better for these isolated and targeted changes to be generalized and grouped into a **broader changeset**, like with the recently merged #24648 . If this isn't possible, another way to effect the change would be to contact one of the contributors and suggest that the change be included with other work that is being done.

The technical reason for rejecting an isolated and targeted change is fairly simple: It is a commit. Yes, git is very good at handling lots of commits, but *humans aren't*. It doesn't help if the commit message history has changes of technical nature mixed with many  others.

All that said, I propose something like the following rules:

- **We do not accept isolated and targeted changes that could be generalized into a larger changeset, unless they are a critical technical concern.**
- **We invite anyone to contribute, so long as no language is used that could be interpreted as politically motivated or otherwise inflammatory.**

If these rules are followed, it **limits exposure** of potentially controversial changes, which allows for a **greater focus on the technical** - *The claimed mission statement of the original rule*.

I would love to include that while *most people don't mind* the phrase "gender neutral", there are *some* who may take offense to it, whether they are judged for it or not. But I fear that using this particular example would go against the purpose of the rule, as it in and of itself can be mischaracterized as inflammatory and "against gender neutral language", which as a linguistic descriptivist is not my stance. If someone can think of a non-hypothetical example that does not invite speculation about the authors character, its inclusion would have great merit.

I hope that the points I made help to create rules that lead to a healthier and truly inclusive community, and I hope I haven't been too ranty.

File: CONTRIBUTING.md (hunk 4 above)
Comment (milesrout) [reply CONTRIBUTING.md:144]: I think it's very cheeky and dismissive of you to assume that you need to give anyone else a 'basic ethics primer'. That kind of attitude is just what people are trying to avoid when they say 'keep politics out of this'. As soon as some people get political they lose all perspective, all respect for others, and all sensible ideas about how other people can be expected to act. 

You don't need a complex rulebook or a big list of rights or rules or an 'ethics primer' to kick out people that are being obnoxious. You just need someone to do it. Anyone that cries 'but the rules are too vague!' instead of 'gosh sorry I really didn't mean to be rude' isn't worth keeping around.

**Comment by n0toose** [comment]
10/10

**Comment by nico** [comment]
I went ahead and merged this. I think it's better than what we had before. We default to merging incremental improvements and iterating in-tree for code changes, so doing the same here seems better than letting stalebot expire this.

**Comment by nico** [comment]
(ps: Meh on the "10/10" comment.)

**Comment by n0toose** [comment]
> (ps: Meh on the "10/10" comment.)

You're right, I took the "stale bot closes PR" the wrong way. I'd like to apologize for being passive aggressive towards you lot in public - I could've asked if this plans to get merged at some point or another or send some sort of a reminder instead.

**Comment by n0toose** [comment]
I suppose that it is especially ironic given that I wrote the line `We may miss the mark sometimes, but we encourage good-faith dialogue over anger.`, but I guess I just missed the mark there myself.

**Comment by nico** [comment]
No big deal, all good :)
